Remove deprecated update_docs

pandu-k opened the pull request:
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Deprecation cleanup

* **What is the current behavior?** (You can also link to an open issue here)
Add_or_update docs endpoint is still available

* **What is the new behavior (if this is a feature change)?**
Removes add_or_update docs endpoint

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Yes. If you are still using add_or_update docs, this will no longer work for you.

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
in progress 

* **Related Python client changes** (link commit/PR here)
[here](https://github.com/marqo-ai/py-marqo/pull/107) 

* **Related documentation changes** (link commit/PR here)
n/a

* **Other information**:


* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

pandu-k committed:
removed references to update_docs

pandu-k committed:
Removed put document tests

pandu-k commented:
Tests passing:
- arm64 API tests: https://github.com/marqo-ai/marqo/actions/runs/5385753028
- unit tests: https://github.com/marqo-ai/marqo/actions/runs/5385751889

vicilliar commented on tests/tensor_search/test_add_documents.py:
```diff
@@ -358,7 +358,7 @@ def test_add_documents_validation(self):
             add_res = tensor_search.add_documents(
                 config=self.config, add_docs_params=AddDocsParams(
                     index_name=self.index_name_1,
-                   docs=bad_doc_arg, auto_refresh=True, update_mode='update', device="cpu"
```
Why was `update_mode='update'` here before? Maybe for previous tests that had that as the update mode, they could have the `use_existing_tensors` flag turned on

vicilliar commented on tests/tensor_search/test_add_documents.py:
```diff
@@ -358,7 +358,7 @@ def test_add_documents_validation(self):
             add_res = tensor_search.add_documents(
```
We could consider removing these tests since the one below testing `replace` covers it

vicilliar commented on tests/tensor_search/test_add_documents.py:
```diff
@@ -402,7 +402,7 @@ def test_add_documents_id_validation(self):
             add_res = tensor_search.add_documents(
```
could be a redundant test as well

vicilliar reviewed (changes requested):
Another note: You also may want to edit the tests in `test_add_documents_use_existing_tensors.py`, as they have `update_mode` set to `replace` as well

pandu-k committed:
removed redundant tests

pandu-k committed:
removed remaining update_mode references

- test_add_documents_use_existing_tensors.py passes
- test_delete_documents.py pass

pandu-k committed:
Merge branch 'mainline' into pandu/remove-upsert

pandu-k commented:
- unit tests: https://github.com/marqo-ai/marqo/actions/runs/5401423357
- CUDA DIND tests: https://github.com/marqo-ai/marqo/actions/runs/5401424359

pandu-k commented:
- unit tests: https://github.com/marqo-ai/marqo/actions/runs/5401445292
- cuda API tests: https://github.com/marqo-ai/marqo/actions/runs/5401446726

pandu-k commented:
> Another note: You also may want to edit the tests in `test_add_documents_use_existing_tensors.py`, as they have `update_mode` set to `replace` as well

Updated these too, and removed other remaining references I found to `update_mode`
